Groq Whisper V3 support

aburkard opened the pull request:
This PR adds support for Groq's Whisper transcription service, which is up to 18x less expensive than OpenAI's hosted model. For English podcasts, the `distil-whisper-large-v3-en` should be good enough since perfect transcription accuracy isn't critical for ad detection.

Also:
- Fixed an FFmpeg issue affecting certain MP3 files.
- Made some updates to the README

aburkard commented on config/config_groq_whisper.yml.example:
```diff
@@ -0,0 +1,45 @@
+  api_key: gsk_XXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  model: whisper-large-v3-turbo
+  language: en
+  # API retry configuration
+  max_retries: 3           # Maximum number of retry attempts for failed API calls
+  initial_backoff: 1.0     # Initial backoff time in seconds before first retry
+  backoff_factor: 2.0      # Multiplier for backoff time between retries
```
The Groq API seems a little flakey sometimes

frrad commented on src/podcast_processor/transcribe.py:
```diff
@@ -158,3 +164,143 @@ def get_segments_for_chunk(self, chunk_path: str) -> List[TranscriptionSegment]:
+        ]
+
+    @staticmethod
+    def add_offset_to_segments(
+        segments: List[GroqTranscriptionSegment], offset_ms: int
+    ) -> List[GroqTranscriptionSegment]:
+        offset_sec = float(offset_ms) / 1000.0
+        for segment in segments:
```
you could refactor to make this function take `List[Segment] -> List[Segment]` and share it with the `RemoteWhisperTranscriber` implementation.

aburkard replied:
I'm a little confused by whether offsets are relative vs. absolute for different segment types, so I'm going to punt on that in this PR.

frrad commented on src/podcast_processor/transcribe.py:
```diff
@@ -158,3 +164,143 @@ def get_segments_for_chunk(self, chunk_path: str) -> List[TranscriptionSegment]:
+        retries = 0
+        backoff_time = self.initial_backoff
+
+        while True:
```
I noticed that there is a `groq` api client [library](https://pypi.org/project/groq/) which would probably give us retries and things for free. Did you consider using that? Slight preference towards using that unless there's a reason not to.

frrad reviewed (commented):
Thanks for working on this! As a user of remote whisper I am very excited about a cheaper alternative. Left a few comments. Also, check out the `ci.sh` script if you haven't seen it already for a way to simulate ci locally.

aburkard commented:
@frrad I removed the changes related to chunk merging that had nothing to do with Groq (https://github.com/jdrbc/podly_pure_podcasts/pull/71/commits/6e9f9df9c4616a4ebd639485c6c0243ed6851e58, https://github.com/jdrbc/podly_pure_podcasts/pull/71/commits/63bdeec7a1bcf77589d8bbf7d898b512914bc82c). I'm going to see if I can recreate the issues I was having with some podcasts separately and will put up a PR if I can figure out what the issue is.

Also I switched to using the Groq python client.

frrad commented:
Hey, sorry for the slow turnaround. This looks good to go once you fix the merge conflicts.
